releng/jenkins/cleanup-project-storage/Jenkinsfile

```diff
@@ -0,0 +1,109 @@
+pipeline {
+  agent {
+    kubernetes {
+      label 'releng-cleanup-nightly-drops-' + env.BUILD_NUMBER
+      defaultContainer 'jnlp'
+      yaml '''
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: jnlp
+    image: 'eclipsecbi/jenkins-jnlp-agent'
+    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
+    resources:
+      limits:
+        memory: "3.6Gi"
+        cpu: "1.0"
+      requests:
+        memory: "3.6Gi"
+        cpu: "1.0"
+    volumeMounts:
+    - mountPath: /home/jenkins/.ssh
+      name: volume-known-hosts
+    - name: settings-xml
+      mountPath: /home/jenkins/.m2/settings.xml
+      subPath: settings.xml
+      readOnly: true
+    - name: m2-repo
+      mountPath: /home/jenkins/.m2/repository
+  volumes:
+  - name: volume-known-hosts
+    configMap:
+      name: known-hosts
+  - name: settings-xml
+    configMap: 
+      name: m2-dir
+      items:
+      - key: settings.xml
+        path: settings.xml
+  - name: m2-repo
+    emptyDir: {}
+    '''
+    }
+  }
+
+  options {
+    buildDiscarder(logRotator(numToKeepStr:'15'))
+    disableConcurrentBuilds()
+    timeout(time: 15, unit: 'MINUTES')
+    timestamps()
+  }
+
+  // https://jenkins.io/doc/book/pipeline/syntax/#triggers
+  triggers {
+    cron('30 21 * * *') // nightly at 21:30
+  }
+  
+  parameters {
+    string      (name: 'NUMBER_OF_DROPS_TO_KEEP', defaultValue: '4', description: 'How many nightly drops should be kept?')
+    booleanParam(name: 'DRY_RUN', description: 'If set the job does not perform file changes')
+  }
+  
+  environment {
+    DOWNLOAD_AREA = '/home/data/httpd/download.eclipse.org/modeling/tmf/xtext/downloads/drops'
+  }
+
+  stages {
+    stage('Cleanup Nightly Drops') {
+      steps {
+        sshagent(['projects-storage.eclipse.org-bot-ssh']) {
+          // Cleanup nightly builds except for the NUMBER_OF_DROPS_TO_KEEP recent ones
+          sh '''
+              # ENV variables and subshells need to be escaped in order to be evaluated remotely.
+              # Exceptions are ENV variables filled by Jenkins locally:
+              #   * NUMBER_OF_DROPS_TO_KEEP
+              #   * DOWNLOAD_AREA
+              #   * DRY_RUN
+							ssh genie.xtext@projects-storage.eclipse.org /bin/bash <<-EOF
```

Comment:
Please convert tabs to spaces

Reply:
We can't do that here without breaking the script! The `-` in `<<-EOF` states that the here-document ignores all tabs (not spaces) in the line of EOF. The alternative would be to remove indentation in the script altogether (or just in the EOF line), but it can't be spaces.

Reply:
Could you please add a comment that it is important do not use spaces here?

Reply:
Done